src/prefect/agent/local/agent.py

```diff
@@ -50,8 +50,16 @@ def deploy_flows(self, flow_runs: list) -> None:
 
             env_vars = self.populate_env_vars(flow_run=flow_run)
 
-            # Pull image if it doesn't exist locally
-            self.docker_client.pull(storage.name)
+            # Check if image exists locally
+            try:
+                self.docker_client.inspect_image(storage.name)
```

Comment:
Hmmmmm I'm on the fence about this.  What about images that are routinely used like `my-flow:latest`?  Seems like we want to allow for re-pulls.  Perhaps a `--no-pull` flag, since I personally think our default should be to pull?

Reply:
I like a no pull flag too